File: 816-design-hashset/design-hashset.cpp
#include <vector>
#include <list>
#include <iostream>

class MyHashSet {
public:
    MyHashSet() {
        // Initialize the hash set with a reasonable size
        table.resize(1000);  // The table size can be adjusted
    }
    
    void add(int key) {
        int index = hash(key);
        // Check if the key already exists in the corresponding bucket
        for (int k : table[index]) {
            if (k == key) return;  // No duplicates allowed
        }
        // Insert the key into the bucket
        table[index].push_back(key);
    }
    
    void remove(int key) {
        int index = hash(key);
        // Find and remove the key from the corresponding bucket
        auto& bucket = table[index];
        for (auto it = bucket.begin(); it != bucket.end(); ++it) {
            if (*it == key) {
                bucket.erase(it);
                return;
            }
        }
    }
    
    bool contains(int key) {
        int index = hash(key);
        // Check if the key exists in the corresponding bucket
        for (int k : table[index]) {
            if (k == key) return true;
        }
        return false;
    }

private:
    std::vector<std::list<int>> table;  // Hash table represented by a vector of lists
    
    // Hash function to map a key to an index
    int hash(int key) {
        return key % table.size();  // Simple hash function
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

File: 816-design-hashset/design-hashset.ts
class MyHashSet {
    private table: number[][];

    constructor() {
        // Initialize the array to represent the hash table.
        this.table = new Array(1000); // You can adjust the size based on performance needs
    }

    // Hash function to compute the index based on the key
    private hash(key: number): number {
        return key % this.table.length;
    }

    add(key: number): void {
        const index = this.hash(key);
        if (!this.table[index]) {
            this.table[index] = [];
        }
        // Avoid duplicate values
        if (!this.table[index].includes(key)) {
            this.table[index].push(key);
        }
    }

    remove(key: number): void {
        const index = this.hash(key);
        const bucket = this.table[index];
        if (bucket) {
            const keyIndex = bucket.indexOf(key);
            if (keyIndex !== -1) {
                bucket.splice(keyIndex, 1); // Remove the key from the bucket
            }
        }
    }

    contains(key: number): boolean {
        const index = this.hash(key);
        const bucket = this.table[index];
        return bucket ? bucket.includes(key) : false;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = new MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */
